test/testers.test.ts

```diff
@@ -0,0 +1,220 @@
+        }
+    };
+    t.false(schemaTypeIs('string')(uiSchema, schema));
+});
+
+test('formatIs should check the format of a resolved sub-schema', t => {
+    const uiSchema: ControlElement = {
+        type: 'Control',
+        scope: {
+            $ref: '#/properties/foo'
+        }
+    };
+    const schema: JsonSchema = {
+        type: 'object',
+        properties: {
+            foo: {
+                type: 'string',
+                format: 'date-time'
+            }
+        }
+    };
+    t.true(formatIs('date-time')(uiSchema, schema));
+});
+
+test('uiTypeIs', t => {
+    const control: ControlElement = {
+        type: 'Control',
+        scope: {
+            $ref: '#/properties/bar'
+        }
+    };
+    t.true(uiTypeIs('Control')(control, undefined));
+});
+
+test('optionIs should check for options', t => {
+    const control: ControlElement = {
+        type: 'Control',
+        scope: {
+            $ref: '#/properties/bar'
+        },
+        options: {
+            answer: 42
+        }
+    };
+    t.true(optionIs('answer', 42)(control, undefined))
+});
+
+test('optionIs should return false for UI schema elements without options field', t => {
+    const control: ControlElement = {
+        type: 'Control',
+        scope: {
+            $ref: '#/properties/bar'
+        }
+    };
+    t.false(optionIs('answer', 42)(control, undefined));
+});
+
+test('schemaMatches should check type sub-schema of control via predicate', t => {
+    const schema: JsonSchema = {
+        type: 'object',
+        properties: {
+            foo: { type: 'string' }
+        }
+    };
+    const uiSchema: ControlElement = {
+        type: 'Control',
+        scope: {
+            $ref: '#/properties/foo'
+        }
+    };
+    t.true(schemaMatches(subSchema => subSchema.type === 'string')(uiSchema, schema));
+});
+
+
+test('schemaMatches should return false for non-control UI schema elements', t => {
+    const schema: JsonSchema = {
+        type: 'object',
+        properties: {
+            foo: { type: 'string' }
+        }
+    };
+    const label: LabelElement = {
+        type: 'Label',
+        text: 'some text'
+    };
+    t.false(schemaMatches(subSchema => false)(label, schema));
+});
+
+test('schemaMatches should return false for control pointing to invalid subschema', t => {
+    const schema: JsonSchema = {
+        type: 'object',
+        properties: {
+            foo: { type: 'string' }
+        }
+    };
+    const uiSchema: ControlElement = {
+        type: 'Control',
+        scope: {
+            $ref: '#/properties/bar'
+        }
+    };
+    t.false(schemaMatches(subSchema => false)(uiSchema, schema));
+});
+
+test('refEndsWith checks whether the ref of a control ends with a certain string', t => {
+    const uiSchema: ControlElement = {
+        type: 'Control',
+        scope: {
+            $ref: '#/properties/bar'
+        }
+    };
+    t.true(refEndsWith('bar')(uiSchema, undefined))
```

Comment:
should this test not show a difference to the refEndIs method, by eg comparing more then the last segment? or less?